### PR TITLE
New Kaomoji Cheat Sheet

### PR DESCRIPTION
**Pull request by irinikp**
## Description of new Instant Answer, or changes
Initializing IA for Kaomoji. This is just a sample to see how an IA is made. It will be updated further. It was made according to instructions from "DuckDuckHack Tutorial: Beginner-level Instant Answers"

---

Instant Answer Page: https://duck.co/ia/view/kaomoji_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by irinikp** [comment]
Updated first comment and json

**Comment by pjhampton** [comment]
Welcome @irinikp 👋

**Comment by irinikp** [comment]
Hi everyone! 👋

File: share/goodie/cheat_sheets/json/kaomoji.json [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        ],
+        "Animals": [
+            {
+                "val": "cat",
+                "key": "(=^･ｪ･^=)"
+            },
+            {
+                "val": "dog",
+                "key": "∪･ω･∪"
+            }
```
Comment (kirkins) [inline share/goodie/cheat_sheets/json/kaomoji.json]: I don't think we can justify the cheat sheet if we only have 6 entries total. Can you expand it with more entries?

Comment (irinikp) [inline share/goodie/cheat_sheets/json/kaomoji.json]: I definitely will in the following days. I just needed to see how a cheat sheet is made and what happens after a pull request.

**Comment by irinikp** [comment]
Does anyone know what this Travis CI build error mean?
traviscibuiltfailed

File: share/goodie/cheat_sheets/json/kaomoji.json [hunk 2]
```diff
@@ -0,0 +1,46 @@
+{
+    "id": "kaomoji_cheat_sheet",
+    "name": "Kaomoji",
+    "description": "Text based emoticons that are created by including Japanese and other 2-byte characters",
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/kaomoji.json]: Lets change description to something like 
'A popular Japanese emoticon style made up of Japanese characters and grammar punctuation.'

Comment (irinikp) [inline share/goodie/cheat_sheets/json/kaomoji.json]: Yeah, that sounds more clear and human friendly

Comment (jophab) [inline share/goodie/cheat_sheets/json/kaomoji.json]: Do consider this while you update pr with more entries :+1:

Comment (irinikp) [inline share/goodie/cheat_sheets/json/kaomoji.json]: Yes I'll replace it in my next PR

**Comment by pjhampton** [comment]
Travis is now fixed. Please rebase and push your work to get the updated build information 👍

**Comment by irinikp** [comment]
Description updated. Travis' checks have passed (thanks @pjhampton ! ).  I will fully populate the emoticons list in a week

File: share/goodie/cheat_sheets/json/kaomoji.json [hunk 3]
```diff
@@ -0,0 +1,451 @@
...
+				"val": "squid",
+				"key": "＜コ：ミ"
+			},
+			{
+				"val": "turtle",
+				"key": ".=.e"
+			},
+			{
+				"val": "whale",
+				"key": ". ><{{{o ______)"
```
Comment (TiagoDanin) [inline share/goodie/cheat_sheets/json/kaomoji.json]: image

Escape!

**Comment by mintsoft** [comment]
@irinikp Are you OK with the escaping issue that's been raised here?

File: share/goodie/cheat_sheets/json/kaomoji.json (hunk 3 above)
Comment (irinikp) [reply share/goodie/cheat_sheets/json/kaomoji.json]: The json is correct, and the { is between "", so I don't see the reason why.
However if you believe removing the {{{ would make a more accurate emoticon, sure I'll change it
(So sorry for the late reply)

Comment (mintsoft) [reply share/goodie/cheat_sheets/json/kaomoji.json]: No problem, however the JSON format here supports an extended syntax including {} if you look through `duckpan server` or on beta you can see the impact. I think if you replace `{`with `\\{` it'll look correct iirc

**Comment by mintsoft** [comment]
Awesome, that's fixed the whale on beta @irinikp :+1: 
@sahildua2305 this LGTM

**Comment by daxtheduck** [comment]
@irinikp this is now merged! It should be live on DuckDuckGo within one week (or during the next release) and we'll be sure to notify you when it happens.

After your Instant Answer goes live for the World to use, you will be able to see [traffic information](http://docs.duckduckhack.com/maintaining/tools.html#monitor-your-instant-answer-page) about your IA as well as receive feedback directly from users. 
                
Please refer to the [maintenance guide and docs](http://docs.duckduckhack.com/maintaining/guidelines.html) to keep up with potential changes to your Instant Answer.
                
Thank you again for your help!

**Pull request merged**